packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -78,9 +76,6 @@ export interface UninitializedVM {
     data: VNodeData;
     /** Shadow Children List */
     children: VNodes;
-    /** Adopted Children List */
-    aChildren: VNodes;
-    velements: VCustomElement[];
```

Comment:
additionally, I think `isRoot` and `state` can be removed.

Reply:
Removed `isRoot`. 
There is one usage of `state` that I am not very clear about. It is used when running `patchShadowRoot()`. patchShadowRoot is invoked when we rehydrate a vm async.
https://github.com/salesforce/lwc/blob/721b37407825477ffc935e06eb1d12b107624d2b/packages/@lwc/engine/src/framework/vm.ts#L342

Will need to think more about this.

Reply:
again, the reason for state to exist was that we were never sure about the current state of the element, so we synthetically track it and adjust when needed, that's not longer the issue. If a routine needs to know the state, it should check for elm.isConnected instead. But I doubt that this is information needed by lwc.

Reply:
Ok, I see this is about `runRenderedCallback`, lets chat more about it.